### PR TITLE
Added new action Get LLM Response

### PR DESCRIPTION
**Pull request by cs-ankit-gupta**
#### Descriptions:
Added new action Get LLM Response

### CODE REVIEW

File: openai/assistant_event_halder.py
Comment (cs-suyog-jinturkar) [inline openai/assistant_event_halder.py:1]: openai/assistant_event_halder.py ---> openai/assistant_event_handler.py

File: openai/assistant_event_handler.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                                                  'content': function_output})
+
+    @override
+    def on_end(self):
+        client = openai.OpenAI(api_key=self.config['apiKey'])
+        run_status = client.beta.threads.runs.retrieve(
+            run_id=self.run_id,
+            thread_id=self.params['threadId']
+        )
+        if run_status.status == 'requires_action':
```
Comment (cs-ankit-gupta) [inline openai/assistant_event_handler.py:98]: Check for run status here

File: openai/assistant_event_handler.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    @override
+    def on_tool_call_delta(self, delta, snapshot):
+        pass
+
+    @override
+    def on_tool_call_done(self, tool_call: ToolCall) -> None:
+        if tool_call.type == 'function':
+            run_status = get_run(config=self.config,
+                                 params={'thread_id': self.params['threadId'], 'run_id': self.run_id})
+            while run_status['status'] in ["queued", "in_progress"]:
```
Comment (cs-ankit-gupta) [inline openai/assistant_event_handler.py:76]: Check for run status here

File: openai/assistant_event_handler.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            )
+        thread_messages = list_thread_messages(config=self.config, params={'thread_id': self.params['threadId']})
+        self.llm_response = thread_messages['data'][0]['content'][0]['text']['value']
+
+    @override
+    def on_exception(self, exception: Exception) -> None:
+        logger.error(f"Exception: {exception}\n", end="", flush=True)
+
+    def get_resposne(self):
+        return self.llm_response
```
Comment (cs-ankit-gupta) [inline openai/assistant_event_handler.py:119]: Do we need this self.response or can we use the on_message_done only

**Review by cs-suyog-jinturkar — APPROVED** [review]
Discussed potential improvements and error handling scenarios. It will be tested and taken care in next PR. Approving the same to unblock QA.

**Pull request merged**